src/test/timedata_tests.cpp

```diff
@@ -34,4 +35,223 @@ BOOST_AUTO_TEST_CASE(util_MedianFilter)
     BOOST_CHECK_EQUAL(filter.median(), 7);
 }
 
+BOOST_AUTO_TEST_CASE(util_MedianFilterShallNotGrowBeyondSize)
+{
+    CMedianFilter<int> filter(2, 15);
+
+    BOOST_CHECK_EQUAL(filter.size(), 1); // 15
+
+    filter.input(100); // 15 100
+    BOOST_CHECK_EQUAL(filter.size(), 2);
+
+    filter.input(10); // 100 10
+    BOOST_CHECK_EQUAL(filter.size(), 2);
+    BOOST_CHECK_EQUAL(filter.sorted()[0], 10);
+    BOOST_CHECK_EQUAL(filter.sorted()[1], 100);
+}
+
+/* Utility function to build a CNetAddr object.
+
+   Using memcpy to build sockaddr_in because compat.h does not support inet_pton with current #define _WIN32_WINNT 0x0501
+*/
+CNetAddr UtilBuildAddress(unsigned char ip1, unsigned char ip2, unsigned char ip3, unsigned char ip4)
+{
+    unsigned char ip[] = {ip1, ip2, ip3, ip4};
+
+    struct sockaddr_in sa;
+    memset(&sa, 0, sizeof(sockaddr_in)); // initialize the memory block
+    memcpy(&(sa.sin_addr), &ip, sizeof(ip));
+    CNetAddr addr = CNetAddr(sa.sin_addr);
+    return addr;
+}
+
+BOOST_AUTO_TEST_CASE(util_UtilBuildAddress)
+{
+    CNetAddr cn1 = UtilBuildAddress(0x001, 0x001, 0x001, 0x0D2); // 1.1.1.210
+
+    CNetAddr cn2 = UtilBuildAddress(0x001, 0x001, 0x001, 0x0D2); // 1.1.1.210
+
+    bool eq = (cn1 == cn2);
+
+    BOOST_CHECK(eq);
+
+    CNetAddr cn3 = UtilBuildAddress(0x001, 0x001, 0x001, 0x0D3); // 1.1.1.211
+
+    bool neq = !(cn1 == cn3);
+
+    BOOST_CHECK(neq);
+
+    neq = !(cn2 == cn3);
+
+    BOOST_CHECK(neq);
+}
+
+BOOST_AUTO_TEST_CASE(util_AddTimeDataAlgorithmComputeOffsetWhenNewSampleCountIsGreaterEqualFiveAndUneven)
+{
+    int capacity = 20; //  can store up to 20 samples
```

Comment:
Should be unsigned to match what `CMedianFilter` expects?

Applies for all `int capacity` in this file.